charts/kubefed/charts/controllermanager/templates/post-install-job.yaml

```diff
@@ -0,0 +1,103 @@
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: "{{ .Release.Name }}-kubefed-config-hook"
+  namespace: "{{ .Release.Namespace }}"
+  annotations:
+    "helm.sh/hook": post-install, post-upgrade
+    "helm.sh/hook-weight": "-5"
+    "helm.sh/hook-delete-policy": hook-succeeded,before-hook-creation
+data:
+  setup.sh: |-
+    #!/bin/bash
+    set -euo pipefail
+
+    kubectl patch kubefedconfig -n {{ .Release.Namespace }} kubefed --type='json' -p='[{"op": "add", "path": "/spec/featureGates", "value":[{"configuration": {{ .Values.featureGates.PushReconciler | default "Enabled" | quote }},"name":"PushReconciler"},{"configuration": {{ .Values.featureGates.CrossClusterServiceDiscovery | default "Disabled" | quote }},"name":"CrossClusterServiceDiscovery"},{"configuration": {{ .Values.featureGates.RawResourceStatusCollection | default "Disabled" | quote }},"name":"RawResourceStatusCollection"},{"configuration": {{ .Values.featureGates.FederatedIngress | default "Disabled" | quote }},"name":"FederatedIngress"},{"configuration": {{ .Values.featureGates.SchedulerPreferences | default "Enabled" | quote }},"name":"SchedulerPreferences"}]}]'
```

Comment:
I'm a bit concerned about this patch. The `op` is `add` for `path` `/spec/featureGates` but that is already present in https://github.com/kubernetes-sigs/kubefed/pull/1346/files#diff-9c32952a289abd6043f1083b7296ab530b31dbbcf0e7ad35f36d61e13393372aR23 if feature gates are specified. What is the behaviour here with `add` operation? I expect that to fail.

Reply:
Ah [JSON patch spec](https://tools.ietf.org/html/rfc6902#section-4.1) details behaviour if field already exists:

> If the target location specifies an object member that does exist, that member's value is replaced.

Reply:
No, that won't fail. In this case, we set the value of `/spec/featureGates`. 

```
If the object targetted by route exists is replaced, if it does not exist it creates it and if it is an Array it creates it in the indicated position in the Array, and the element “-” is used to add new elements to an Array.
```
We re-create the value of featureGates here to my understanding.
```
$ kubectl logs -f kubefed-7qzkgwlapd-b5rt8 -n kube-federation-system2
kubefedconfig.core.kubefed.io/kubefed patched
Kubefedconfig patched successfully!
```